### ISSUE TITLE
[Bug]: Incognito browser context doesn't respect proxyServer option

### ISSUE THREAD

**Issue by vickyRathee** [puppeteer/puppeteer#8820]
### Bug description

Steps to reproduce the problem:

## Example 1
Launch browser with proxy server
```
return await puppeteer.launch({
      headless: headless,
      executablePath: CHROMIUM_PATH,
      ignoreHTTPSErrors: true,
      args: [
        `--proxy-server=${process.env.PROXY_SERVER}`,
        "--no-sandbox",
        "--disable-setuid-sandbox",
      ],
    });
```

Create a context without proxy
```
const context = await browser.createIncognitoBrowserContext({
              proxyServer: undefined,
            });
const page = await context.newPage();
await page.goto("https://example.com");

-- Throw error : proxy authentication required

```
I want to skip [proxy inherited](https://github.com/puppeteer/puppeteer/commit/3feca325a9472ee36f7e866ebe375c7f083e0e36) from browser instance, so set the `proxyServer: undefined` for this context.

But the context page still asking for proxy authentication, which means it's not respecting the `proxyServer` option set in create and just inheriting it from the browser anonymously.

## Example 2

It doesn't work vice-versa as well, 

Launch browser without proxy server

```
return await puppeteer.launch({
      headless: headless,
      executablePath: CHROMIUM_PATH,
      ignoreHTTPSErrors: true,
      args: [
        "--no-sandbox",
        "--disable-setuid-sandbox",
      ],
    });
```

Create a context with proxy
```
const context = await browser.createIncognitoBrowserContext({
              proxyServer: process.env.PROXY_SERVER,
            });
const page = await context.newPage();
await page.goto("https://example.com");
```

Same goes vice-versa.  I am setting the `proxyServer` option set in create context, but it's not setting the proxy and inheriting from browser level causing no proxy applied.

So **we are forced to launch 2 browser** 

1. With proxy in launch argument
2. Without proxy in launch argument

Due to the `proxyServer` option is useless in creating context.

### Puppeteer version

16.2.0

### Node.js version

16.14.0

### npm version

7.20.0

### What operating system are you seeing the problem on?

Windows

**Comment by OrKoN** [comment]
It does sound a lot like https://crbug.com/1310057 which is an upstream issue in Chromium, that does not respect the proxy settings on Windows.

**Comment by vickyrathee** [comment]
@OrKoN What about vice-versa, is that also related to windows only due to upstream?

I mean - when I don't set proxy in launch arguments but trying to use proxy in context by passing `proxyServer`

**Comment by OrKoN** [comment]
@vickyrathee yes, I believe the proxyServer param in createIncognitoBrowserContext is completely ignored on Windows.

**Comment by sequencerr** [comment]
Not working on Linux(Debian) (vmware) for me either. https://github.com/puppeteer/puppeteer/issues/678#issuecomment-1214189656
```js
const context = await browserInstance.createIncognitoBrowserContext({
	proxyServer: 'http://***:80'
});
const page = await context.newPage();
await page.authenticate({ username: '***', password: '***' });
```

**Comment by OrKoN** [comment]
The upstream issue has been fixed so this should be fixed in M129 (https://crbug.com/40830145)

**Comment by Lightning00Blade** [comment]
Chrome 129.0.6668.58 landed with the release of [Puppeteer v23.4.0](https://github.com/puppeteer/puppeteer/blob/puppeteer-v23.4.0/docs/api/index.md) I am closing this issue.
